core/src/main/scala/org/bitcoins/core/psbt/PSBTMap.scala

```diff
@@ -621,6 +621,49 @@ case class InputPSBTMap(elements: Vector[InputPSBTRecord])
 object InputPSBTMap extends PSBTMapFactory[InputPSBTRecord, InputPSBTMap] {
   import org.bitcoins.core.psbt.InputPSBTRecord._
 
+  def fromWitness(witness: ScriptWitness): InputPSBTMap = {
+    witness match {
+      case p2wpkh: P2WPKHWitnessV0 =>
+        val sig = PartialSignature(p2wpkh.pubKey, p2wpkh.signature)
+        InputPSBTMap(Vector(sig))
+      case p2wsh: P2WSHWitnessV0 =>
+        val sigs = p2wsh.pubKeys.zip(p2wsh.signatures).map {
```

Comment:
same as above, this may not have the signatures yet. either way, you also want to put the witness script record in